src/machine/CoffeeMachineCommandsScanner.java

```diff
@@ -0,0 +1,35 @@
+package machine;
+
+import java.util.Scanner;
+
+class CoffeeMachineCommandsScanner {
+
+    private static void askForAction(CoffeeMachine coffeeMachine) {
+        Scanner scanner = new Scanner(System.in);
+        System.out.println("Write action (buy, fill, take, remaining, exit):");
+        String input = scanner.nextLine();
+        coffeeMachine.processUserInput(input);
+    }
+
+    private static void askForTypeOfCoffee(CoffeeMachine coffeeMachine) {
+        Scanner scanner = new Scanner(System.in);
```

Comment:
We could reuse the same `Scanner` instance instead of creating a new one every time